src/controllers/token.rs

```diff
@@ -286,28 +283,17 @@ pub async fn revoke_current_api_token(app: AppState, req: Parts) -> AppResult<Re
     Ok(StatusCode::NO_CONTENT.into_response())
 }
 
-struct NewTokenEmail<'a> {
-    token_name: &'a str,
-    user_name: &'a str,
-    domain: &'a str,
-}
-
-impl crate::email::Email for NewTokenEmail<'_> {
-    fn subject(&self) -> String {
-        format!("crates.io: New API token \"{}\" created", self.token_name)
-    }
-
-    fn body(&self) -> String {
-        format!(
-            "\
-Hello {user_name}!
-
-A new API token with the name \"{token_name}\" was recently added to your {domain} account.
-
-If this wasn't you, you should revoke the token immediately: https://{domain}/settings/tokens",
-            token_name = self.token_name,
-            user_name = self.user_name,
-            domain = self.domain,
-        )
-    }
+async fn send_creation_email(
+    emails: &crate::Emails,
+    recipient: &str,
+    token_name: &str,
+    user_name: &str,
+) -> anyhow::Result<()> {
+    let context = context! {
+        token_name,
+        user_name,
+        domain => emails.domain,
+    };
+    let email = EmailMessage::from_template("new_token", context)?;
+    Ok(emails.send(recipient, email).await?)
```

Comment:
Nit: Why is the template context not built outside this function and passed into it, as is done with others, but rather constructed within this function?

Reply:
Nit: Same here, `.context()` with either rendering error or a sending error.